satpy/writers/cf_writer.py

```diff
@@ -925,3 +937,39 @@ def save_datasets(self, datasets, filename=None, groups=None, header_attrs=None,
             written.append(res)
 
         return written
+
+
+def _check_backend_versions():
+    """Issue warning if backend versions do not match."""
+    if not _backend_versions_match():
+        warnings.warn(
+            "Backend version mismatch. Compression might fail or be ignored "
+            "silently. Recommended: All versions below or above "
+            "netCDF4-1.6.0/libnetcdf-4.9.0/xarray-2022.12.0.",
+            stacklevel=2
```

Comment:
With 3 I think this would show the caller of the writer's `__call__` method?

```suggestion
            stacklevel=3
```